server.go

```diff
@@ -1944,6 +1961,40 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets a compressor for outbound messages.
+// It must not be called after any event that causes headers to be sent
+// (see SetHeader for a complete list). Provided compressor is used when below
+// conditions are met:
+//
+//   - compressor is registered via encoding.RegisterCompressor
+//   - compressor name exists in the client advertised compressor names sent in
+//     :grpc-accept-encoding header.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// The error returned is compatible with the status package.  However, the
+// status code will often not match the RPC status as seen by the client
+// application, and therefore, should not be relied upon for this purpose.
+// It is not safe to call SetSendCompressor concurrently with SendHeader and
+// SendMsg.
+//
+// # Experimental
+//
+// Notice: This type is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func SetSendCompressor(ctx context.Context, name string) error {
+	stream, ok := ServerTransportStreamFromContext(ctx).(*transport.Stream)
+	if !ok || stream == nil {
+		return status.Errorf(codes.Internal, "grpc: failed to fetch the stream from the context %v", ctx)
```

Comment:
Is it common to log the context with `%v`?  If there's truly going to be useful information in there then LGTM but I'm not sure how it looks TBH and we should remove it if it's too cluttered or just prints a pointer or something.

Reply:
It does look very cluttered with all the fields being printed inline. Something like:
```
context.Background.WithValue(type string, val test-val)
```
Also, I don't think this can highlight any meaningful information. Dropped context from the error message